### PR TITLE
OPMPHM: add random and hash function

### PR DESCRIPTION
**Pull request by KurtMi**
Should do #887 add {random,hash} function

### CODE REVIEW

File: src/include/kdbopmphm.h [hunk 1]
```diff
@@ -1,6 +1,55 @@
 #ifndef OPMPHM_H
 #define OPMPHM_H
 
+/* Hash function
+ * By Bob Jenkins, May 2006
+ * http://burtleburtle.net/bob/c/lookup3.c
+ */
+
+#define rot(x, k) (((x) << (k)) | ((x) >> (32 - (k))))
```
Comment (KurtMi) [inline src/include/kdbopmphm.h]: I copied only the code, I will investigate and change it.

File: src/libs/elektra/opmphm.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/**
+ * @file
+ *
+ * @brief The Order Preserving Minimal Perfect Hash Map.
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+
+#include "kdbprivate.h"
```
Comment (KurtMi) [inline src/libs/elektra/opmphm.c]: I have some questions regarding the reusability of my code, I would leave it so for now and discuss my questions the next time we meet.

**Comment by KurtMi** [comment]
> Is it really necessary to have macros here? Did you benchmark that (inline) functions are slower?

I made some benchmarks and found out that inlining is in my opinion to slow.

Open things raised by this PR are noted in #887.

**Comment by KurtMi** [comment]
@markus2330 this PR would be ready, if you don't have any objection.

**Comment by markus2330** [comment]
Thanks great job! Please also set "ready to merge" label next time, there are plenty of ways to misunderstand "PR would be ready" (ready to review? OR: it would be, but it is not)

**Pull request merged**